lib/token-request.js

```diff
@@ -321,7 +321,7 @@ TokenRequest.prototype._getTokenUsernamePasswordFederated = function(username, p
       return;
     }
 
-    this._performUsernamePasswordForAccessTokenExchange(this._userRealm.federationActiveAuthUrl, username, password, callback);
+    this._performUsernamePasswordForAccessTokenExchange(this._userRealm.federationActiveAuthUrl, constants.WSTrustVersion.UNDEFINED, username, password, callback);
```

Comment:
Hmm, I don't think this should be undefined. I think we can parse the WSTrustVersion from the federation ActiveAuthUrl. For example, on my ADFS I get federation_active_auth_url: "https://fs.ajmichael.net/adfs/services/trust/2005/usernamemixed". I think adal can parse this from the endpoint. Looking at my ADFS Service manager, I do not see any way to customize the endpoint names (although there may be) so I think that it is safe to regex the url.